scripts/inject_fault.tcl

```diff
@@ -576,58 +576,201 @@ proc select_random_net {} {
 #  Flip a Bit  #
 ################
 
-# flip a spefific bit of the given net name. returns a 1 if the bit could be flipped
+# flip a spefific bit of the given net name. returns [success, old_value, new_value] if the bit could be flipped
 proc flipbit {signal_name is_register} {
-  set success 0
-  set old_value [examine -radixenumsymbolic $signal_name]
-  # check if net is an enum
-  if {[examine -radixenumnumeric $signal_name] != [examine -radixenumsymbolic $signal_name]} {
-    set examine_string [examine -unsigned -radixenumnumeric $signal_name]
-    regexp {(\d*)'\w(\d*)} $examine_string -> length old_value_numeric
-    if {$length != 1} {
-      set new_value_numeric [expr int(rand()*([expr 2 ** $length]))]
-      while {$old_value_numeric == $new_value_numeric} {
-        set new_value_numeric [expr int(rand()*([expr 2 ** $length]))]
+  # Get the current value in binary format e.g. "6'b101010" - this works both for enums and normal signals
+  set old_value_string [examine -radixenumnumeric -binary -showbase $signal_name]
```

Comment:
Same here you should use the -`showbase` flag, otherwise it is possible that the length of the signal cannot be correctly extracted.